pkg/config/config.go

```diff
@@ -49,26 +52,36 @@ func New(filename string) (*Config, error) {
 	if cfg.PrometheusPort == 0 {
 		cfg.PrometheusPort = defaultPrometheusPort
 	}
+	if debug {
+		cfg.Debug = debug
+	}
 
 	return &cfg, nil
 }
 
 func (c Config) Print() {
-	if c.StagingDir != "" {
-		log.Printf("Staging directory: %q", c.StagingDir)
-	} else {
-		log.Printf("No staging directory. Writing directly to the archive dir")
+	ops := [][]string{
```

Comment:
That was my initial choice, but map does not preserve the order and I don't want help screen to be sorted by random()

Reply:
but the current implementation along with `ops = ops[1:]` is ugly

Reply:
IMO it's less ugly than a bunch of unstructured log.Printfs, although I'll think about improving it.

Reply:
TBH I don't see a problem with bunch of `logger.G.With(<param>, <value>).Info("option")`

Reply:
what you really can do to improve that part, is to make use of `reflect` to iterate over config struct fields

Reply:
The problem with a bunch of logger statements is that they promote unnecessary repetitions and shift the solution to writing more code instead of using proper data structures . The list solution is adequate, albeit some ugly parts you've highlighted; the reflection would neither solve the issue (as we don't output the literal name of the parameter, but some human-readable variation of it), nor simplify the code. Instead, I'm changing it back to maps and just sorting by keys to ensure a consistent output on every run.

Reply:
Turns out map approach doesn't buy us much in terms of readability, as it is not possible to write a function like ```ternary(ptr != nil, fmt.Sprintf("%v", *ptr), "")``` without crashing if ptr is null, reverted back to just show each parameter in a separate log.